lib/puma/dsl.rb

```diff
@@ -1270,6 +1270,37 @@ def fork_worker(after_requests=1000)
       @options[:fork_worker] = Integer(after_requests)
     end
 
+    # When enabled, workers will be converted to a "mold" process as needed from which further workers
+    # are forked. This option is similar to fork_worker but the process that does the reforking does
+    # not take any further traffic. This allows the mold process to be optimized for copy-on-write
+    # performance and stability.
+    #
+    # This option also enables the `refork` command (SIGURG), which allows external processes to trigger
+    # promotion and reforking from a new mold process.
+    #
+    # Reforks will trigger automatically as workers hit the specified number of requests (default 1000),
+    # and multiple intervals can be specified (as absolute request count thresholds) to allow for improved
+    # performance over time.
+    def mold_worker(mold_at=1000, *additional_molds)
+      @options[:mold_worker] = [Integer(mold_at)] + additional_molds.map { |m| Integer(m) }
```

Comment:
I think we should validate here that `fork_worker` hasn't also been enabled, and vice versa. We should also do so in the CLI if possible.

Reply:
we get validation for free in the CLI by implementing it in the DSL. I think the right approach is to just ignore the second config if someone tries to activate both and log a warning, which I'll go ahead and do; doesn't look like any existing configs fail aggressively on conflict.